pkg/storaged/btrfs/utils.jsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import { decode_filename } from "../utils.js";
+
+/*
+ * Calculate the usage based on the data from `btrfs filesystem show` which has
+ * been made available to client.uuids_btrfs_usage. The size/usage is provided
+ * per block device.
+ */
+export function btrfs_device_usage(client, uuid, path) {
+    const block = client.blocks[path];
+    const device = block && block.Device;
+    const uuid_usage = client.uuids_btrfs_usage[uuid];
+    if (uuid_usage && device) {
+        const usage = uuid_usage[decode_filename(device)];
+        if (usage) {
+            return [usage, block.Size];
+        }
+    }
+    return [0, block.Size];
+}
+
+/**
+ * Calculate the overal btrfs "volume" usage. UDisks gives us the overal usage the blocks the space.
+ */
+export function btrfs_usage(client, volume) {
+    const block_fsys = client.blocks_fsys[volume.path];
+    const mount_point = block_fsys && block_fsys.MountPoints[0];
+    let use = mount_point && client.fsys_sizes.data[decode_filename(mount_point)];
+    if (!use)
+        use = [volume.data.used, client.uuids_btrfs_blocks[volume.data.uuid].reduce((sum, b) => sum + b.Size, 0)];
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20231214-115043-fd396f8d-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)